src/style-spec/validate/validate_layer.js

```diff
@@ -71,10 +71,14 @@ module.exports = function validateLayer(options) {
         }
     }
 
+    // https://github.com/mapbox/mapbox-gl-js/issues/5772
+    const valueSpec = extend({}, styleSpec.layer,
+        {type: extend({}, styleSpec.layer.type, {required: false})});
```

Comment:
This is a little on the hacky side for my taste...

What do you think about adding a no-op function for `type` in `objectElementValidators` instead?

Reply:
It can't be a no-op because we need to validate the other schema properties, and it requires an additional modification, so I think it comes out as a wash in terms of hackiness. Happy to go with either approach though -- take a look at see which one you prefer.